llvm/lib/Transforms/IPO/SampleProfile.cpp

```diff
@@ -444,15 +444,36 @@ class SampleProfileMatcher {
   // the profile.
   StringMap<LocToLocMap> FuncMappings;
 
-  // Profile mismatching statstics.
+  // Match state for an anchor/callsite.
+  enum class MatchState {
+    InitialMatch = 0,
+    InitialMismatch = 1,
+    // From initial mismatch to final match.
+    RecoveredMismatch = 2,
+    // From initial match to final mismatch.
+    RemovedMatch = 3,
+    Unknown = 32,
```

Comment:
Usually we use 0 as uninitialized/unknown value. This is because for zero initialized memory, this naturally sets everything to unknown.